doc/whats_new.rst

```diff
@@ -56,6 +56,9 @@ Enhancements
      :class:`linear_model.LogisticRegression`, by avoiding loss computation.
      By `Mathieu Blondel`_ and `Tom Dupre la Tour`_.
 
+   - Improved heuristic for ``class_weight="auto"`` for classifiers supporting
+     ``class_weight`` by Hanna Wallach and `Andreas Müller`_
```

Comment:
There should be a bit more details here: at least explain that `class_weight="auto"` has been deprecated in favor of `class_weight="balanced"` that provides invariance to class imbalance introduced by re-sampling with replacement for linear models.

Maybe also mention `class_weight='balanced_subsample'` for forest models.